packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -238,40 +238,11 @@ function serializeClassAttribute(part: VStaticPartElement, classToken: string) {
     return computedClassName.length ? ` class="${htmlEscape(computedClassName, true)}"` : '';
 }
 
-const enum FragmentCache {
-    HAS_SCOPED_STYLE = 1,
-    SHADOW_MODE_SYNTHETIC = 2,
-}
-
-// This should be a no-op outside of LWC's Karma tests, where it's not needed
-let registerFragmentCache: (fragmentCache: any) => void = noop;
-
-// Only used in LWC's Karma tests
-if (process.env.NODE_ENV === 'test-karma-lwc') {
-    // Keep track of fragmentCaches, so we can clear them in LWC's Karma tests
-    const fragmentCaches: any[] = [];
-    registerFragmentCache = (fragmentCache: any) => {
-        fragmentCaches.push(fragmentCache);
-    };
-
-    (window as any).__lwcResetFragmentCaches = () => {
-        for (const fragmentCache of fragmentCaches) {
-            for (const key of keys(fragmentCache)) {
-                delete fragmentCache[key];
-            }
-        }
-    };
-}
-
 function buildParseFragmentFn(
     createFragmentFn: (html: string, renderer: RendererAPI) => Element
 ): (strings: string[], ...keys: (string | number)[]) => () => Element {
-    return (strings: string[], ...keys: (string | number)[]) => {
-        const cache = create(null);
```

Comment:
Note this cache was effectively tied to every tagged template literal. Each tagged template literal has a unique `string[]` object, so using the `string[]` as the cache key in a WeakMap effectively gives us a unique-per-tagged-template-literal cache.